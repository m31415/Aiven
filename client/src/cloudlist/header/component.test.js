import CloudListHeaderComponent from "./component";
import React from "react";
import { shallow } from "enzyme";

describe("CloudListHeaderComponent", () => {
  const lang = "de-DE";
  const providerFilter = "do";
  const sortDistance = true;

  let useContextSpy;
  let setProviderFilter;
  let setSortDistance;
  let tree;

  beforeEach(() => {
    setProviderFilter = jest.fn();
    setSortDistance = jest.fn();

    const testContext = {
      lang,
      providerFilter,
      setProviderFilter,
      sortDistance,
      setSortDistance
    };

    useContextSpy = jest.spyOn(React, "useContext");
    useContextSpy.mockImplementation(() => testContext);
    tree = shallow(<CloudListHeaderComponent />);
  });

  it("renders as usual", () => {
    expect(tree.debug()).toMatchSnapshot();
  });

  it("Button recieves correct variant if sortDistance", () => {
    const variant = tree.find({ name: "sortDistance" }).prop("variant");
    expect(variant).toEqual("warning");
  });

  it("Button toggles sortDistance onClick", () => {
    tree.find({ name: "sortDistance" }).simulate("click");
    expect(setSortDistance).toBeCalledWith(false);
  });

  it("Dropdown uses Provider dict to map keys to name", () => {
    const name = tree.find({ eventKey: "google" }).text();
    expect(name).toEqual("Google Cloud");
  });

  it("Dropdown Item select sets providerFilter", () => {
    tree.find("DropdownButton").simulate("select", "google");
    expect(setProviderFilter).toBeCalledWith("google");
  });
});
