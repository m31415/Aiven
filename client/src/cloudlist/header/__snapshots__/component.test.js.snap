// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudListHeaderComponent renders as usual 1`] = `
"<div className=\\"row cloudListHeader background text-center\\">
  <div>
    <h1 className=\\"headerText\\">
      Cloud Selection
    </h1>
  </div>
  <div className=\\"row filter\\">
    <p className=\\"col\\">
      Sort by Distance
    </p>
    <div className=\\"col\\">
      <Button name=\\"sortDistance\\" variant=\\"warning\\" onClick={[Function: onClick]} active={false} disabled={false} type=\\"button\\">
        <i className=\\"fas fa-location-arrow\\" />
      </Button>
    </div>
  </div>
  <div className=\\"row filter\\">
    <p className=\\"col\\">
      Filter by Provider
    </p>
    <div className=\\"col\\">
      <DropdownButton alignRight={true} id=\\"dropdown-menu-align-right\\" variant=\\"warning\\" onSelect={[Function: select]} title=\\"Digital Ocean\\">
        <DropdownItem eventKey={{...}} as={{...}} disabled={false}>
          All
        </DropdownItem>
        <DropdownItem eventKey=\\"aws\\" as={{...}} disabled={false}>
          Amazon Web Services
        </DropdownItem>
        <DropdownItem eventKey=\\"do\\" as={{...}} disabled={false}>
          Digital Ocean
        </DropdownItem>
        <DropdownItem eventKey=\\"google\\" as={{...}} disabled={false}>
          Google Cloud
        </DropdownItem>
        <DropdownItem eventKey=\\"azure\\" as={{...}} disabled={false}>
          Azure
        </DropdownItem>
        <DropdownItem eventKey=\\"upcloud\\" as={{...}} disabled={false}>
          UpCloud
        </DropdownItem>
      </DropdownButton>
    </div>
  </div>
</div>"
`;
